object/path.go

```diff
@@ -0,0 +1,26 @@
+package object
+
+import "strings"
+
+// Directories returns the directories for the given path. When explicit is
+// true, the returned directories do not include the path itself should it be a
+// directory. The root path ('/') is always excluded.
+func Directories(path string, explicit bool) (dirs []string) {
```

Comment:
@ChrisSchinnerl I'm not a big fan of the naming here, I went with `explicit` because we have this test called `TestObjectsExplicitDir` re-reading it now makes me think I have to inverse it actually. Holding off on that for now until we review the entire thing.

Reply:
Tbh I'm not a fan of having the bool. It feels a bit too forced. The way I understand it is that it's only required for `InsertDirectoriesForRename`. Looking at that method it seems very verbose. I'm pretty sure your first instinct was to just update the directory names instead of doing the whole mapping thing but you ran into issues due to the fact that the `directories` table doesn't contain the bucket.

I think it might be worth adding the bucket id to the directories table and extending the unique key to cover that.
Then simplify `InsertDirectoriesForRename` and finally get rid of the `explicit` bool since we should be able to just update all directories of the bucket that we rename within. My guess is that that's also a lot faster than looping and inserting directories.

That makes the migration more complicated but if we ever decide to use the `directories` table as a cache for fields like `size` and `health` of contained objects, we will need to introduce buckets to that table anyway.